translate to x86_64 with msabi appears to be untested

(at least) the pointer args passed to the generated code must be moved as 64-bit quantities

--- origsrc/Mesa-8.0.5/src/gallium/auxiliary/translate/translate_sse.c	2012-10-24 20:03:59.000000000 +0100
+++ src/Mesa-8.0.5/src/gallium/auxiliary/translate/translate_sse.c	2013-10-03 19:38:14.786600000 +0100
@@ -1299,14 +1299,22 @@ static boolean build_vertex_emit( struct
    x86_push(p->func, p->outbuf_EBX);
    x86_push(p->func, p->count_EBP);
 
-/* on non-Win64 x86-64, these are already in the right registers */
+   /* on non-Win64 x86-64, these are already in the right registers */
    if(x86_target(p->func) != X86_64_STD_ABI)
    {
       x86_push(p->func, p->machine_EDI);
       x86_push(p->func, p->idx_ESI);
 
-      x86_mov(p->func, p->machine_EDI, x86_fn_arg(p->func, 1));
-      x86_mov(p->func, p->idx_ESI, x86_fn_arg(p->func, 2));
+      if(x86_target(p->func) != X86_32)
+      {
+        x64_mov64(p->func, p->machine_EDI, x86_fn_arg(p->func, 1));
+        x64_mov64(p->func, p->idx_ESI, x86_fn_arg(p->func, 2));
+      }
+      else
+      {
+        x86_mov(p->func, p->machine_EDI, x86_fn_arg(p->func, 1));
+        x86_mov(p->func, p->idx_ESI, x86_fn_arg(p->func, 2));
+      }
    }
 
    x86_mov(p->func, p->count_EBP, x86_fn_arg(p->func, 3));
